File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
         if(x<0)
{
    return 0;
}
    long long int  copy, rev=0,rem;
    
        copy=x;
         while (copy!=0)
         {
             rem = copy % 10;
            rev= rev*10+rem;
            copy= copy/10;
         }
         return (x==rev);
    }

};

File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if ( x<0|| x%10 ==0 && x!= 0 )
        {
            return false;
        }
         int revHalf= 0 ;
         while( revHalf<x )
         {
int rem = x%10;
revHalf = revHalf *10+rem;
 x=x/10; 
         }
          return (x==revHalf || x==revHalf/10);
    }
}